### ISSUE TITLE
Deductions not showing rows for all itinerary dates

### ISSUE THREAD

**Issue by janpanchalingam1** [unicef/etools-issues#402]
Browser: IE and Chrome
Workspace: Sierra Leone

1) Traveller saves a trip for 1 day
2) Traveller updates itinerary for 3 days and submits to Supervisor for approval
3) Deductions not updating for selected itinerary dates

**Comment by janpanchalingam1** [comment]
![image](https://cloud.githubusercontent.com/assets/26144740/26731801/39eaa7bc-4783-11e7-820e-6f1877a400e7.png)

**Comment by janpanchalingam1** [comment]
![sharedimage](https://cloud.githubusercontent.com/assets/26144740/26732088/5e283d00-4784-11e7-96b7-071e0341f2bf.png)

**Comment by torbent** [comment]
@kasfaw @janpanchalingam1 what is the correct behaviour here?
I understand that the UX looks a bit misleading, but according to the documentation, we're always handling the last day the same, meaning the last days deductions is always 40% for everything, and showing deductions for that day would be equally misleading..
The "last day" is defined as "the last departure", which is why the 10th ( in the example ) is not shown either... attaching the flowchart for calculating deduction and DSA:

**Comment by torbent** [comment]
![DSA_and _Deductions_diagram.png](https://waffleio-direct-uploads-production.s3.amazonaws.com/uploads/58ca6974c9b0d72000667588/125516c66e82c728ace21e0d46cca7bb1c838dadcfd69b46a048f380204929601600aa2a7797e225fd0e39111d0014f9121c4d4eeefc8179a1b23d248c5e5fe8826a5720c73d10feb0d44ebc00a83c990dd4e8e8d51199a26867f5819d4c3c14041c80f85936a8904ecc51f349f300.png)

**Comment by lordfuoco** [comment]
@janpanchalingam1 @kasfaw keeping in mind #59 (timezones ) we went trough this once again, and we have a better understanding of the problem ( which we do not experience too much since we are closer to UTC time ):

The last departure date is included in the deductions list ( up to 40% )
![Screen Shot 2017-06-15 at 16.07.52.png](https://waffleio-direct-uploads-production.s3.amazonaws.com/uploads/58d12f6a9179201a005e04c3/125516c66e82c728ace21e0d46db978826878de9bce1b057e40fa0c57e0b77256455f66b7191a379e90c2b56415a4cbe5d4a1f06a3a3dc78e2a436628f5459ead56c0b72c46914ffbf8a4bbe01bb728319d4b5bd9850d2aa6c7bb3d4d216744b470a83ed5e37aa984fc15ff348e44b5ca211316bf43b.png)

If is not included it may be for 2 reasons: 

1) the date is exactly at midnight, in that case the system do not include the last day:
![Screen Shot 2017-06-15 at 16.09.09.png](https://waffleio-direct-uploads-production.s3.amazonaws.com/uploads/58d12f6a9179201a005e04c3/125516c66e82c728ace21e0d46db978826878de9bce1b057e40fa0c57e0b77256455f66b7191a379e90c2b58415f47be5d4a1f06a3a3dc78e2a436628f5459ead56c0b72c46914ffbf8a4bbe01bb728319d4b5bd9850d2aa6c7bb3d4d216744b470a83ed5e37aa984fc15ff348e44a53a0173462f43b.png)
(This is the behaviour that we had till now, unchanged in the last 3 months -  actually from before February ) We can change it if is needed.

2) the trip is saved in the backend ( as per jan screenshots ) and the date even if not midnight is on a timezone that - when saved in the backend - would send the day till behind  - or exactly on - the midnight of the day.
When the trip is saved the frontend do not calculate dates because the right policy is to 'respect the backend'

In the above screenshot so the timezone offset is bringing the date on the midnight or before it when the trip is sent to the backend, I fired a browser with a fake timezone placing myself on -7 and this is the result with last Departure date at 4:00AM (error)  ->
![Screen Shot 2017-06-15 at 16.25.15.png](https://waffleio-direct-uploads-production.s3.amazonaws.com/uploads/58d12f6a9179201a005e04c3/125516c66e82c728ace21e0d46db978826878de9bce1b057e40fa0c57e0b77256455f66b7191a379e90c2954415e4bbe5d4a1f06a3a3dc78e2a436628f5459ead56c0b72c46914ffbf8a4bbe01bb728319d4b5bd9850d2aa6c7bb3d4d216744b470a83ed5f33aa984fc15ff348e74a56a5133265f43b.png)

And this is the same trip  with 8:00 AM (OK) ->
![Screen Shot 2017-06-15 at 16.26.00.png](https://waffleio-direct-uploads-production.s3.amazonaws.com/uploads/58d12f6a9179201a005e04c3/125516c66e82c728ace21e0d46db978826878de9bce1b057e40fa0c57e0b77256455f66b7191a379e90c2957415f4ebe5d4a1f06a3a3dc78e2a436628f5459ead56c0b72c46914ffbf8a4bbe01bb728319d4b5bd9850d2aa6c7bb3d4d216744b470a83ed5f33aa984fc15ff348e74a52a01b3e63f43b.png)

So this is a timezone issue and eventually the midnight edge case. I hope I was detailed enough.

For the last screenshot I used the same trip mentioned here in the issue and a timezone set on 'US/Pacific'
